rdeprojector/controllers/rdeprojector_controller.go

```diff
@@ -87,7 +107,75 @@ func (r *RDEProjectorReconciler) reconcileNormal(ctx context.Context, rdeProject
 	capiYaml := entity["spec"].(map[string]interface{})["capiYaml"]
 	log.Info("Retrieved Capi Yaml", "capiYaml", capiYaml)
 
-	//TODO Apply CAPI YAML
+	// Retrieve kubeconfig of the cluster
+	kubeconfig := ctrl.GetConfigOrDie()
+
+	// Prepare a RESTMapper to find GVR
+	dc, err := discovery.NewDiscoveryClientForConfig(kubeconfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+
+	// Get the dynamic client
+	dyn, err := dynamic.NewForConfig(kubeconfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+	// Prepare the Yaml Reader
+	yamlReader := k8syaml.NewYAMLReader(bufio.NewReader(bytes.NewReader([]byte(fmt.Sprintf("%v", capiYaml)))))
+
+	for err == nil {
+		// Read single API object at a time from the CAPI Yaml
+		yamlBytes, err := yamlReader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "Error reading object from the CAPI yaml of rdeId [%s]; yaml bytes: [%v]", rdeProjector.Spec.RDEId, yamlBytes)
+		}
+
+		// Decode the yaml API object into unstructured format
+		obj := &unstructured.Unstructured{}
+		_, gvk, err := decUnstructured.Decode(yamlBytes, nil, obj)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "Error decoding CAPI object from the CAPI yaml of rdeId [%s] into unstructured format; yaml bytes: [%v] ", rdeProjector.Spec.RDEId, yamlBytes)
+		}
+
+		// Get GVR (Group, Version, Resource) of the object
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "Error constructing GVR of capi object retrieved from rdeId [%s]; gvk: [%v]", rdeProjector.Spec.RDEId, gvk)
+		}
+
+		// Get REST client/interface for the given GVR of the object
+		var dr dynamic.ResourceInterface
+		if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
+			// namespaced resources should specify the namespace
+			dr = dyn.Resource(mapping.Resource).Namespace(obj.GetNamespace())
+		} else {
+			// for cluster-wide resources
+			dr = dyn.Resource(mapping.Resource)
+		}
+
+		// Marshal object into the JSON
+		data, err := json.Marshal(obj)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "Error marshaling capi object into JSON format; rdeId: [%s]; capi object: [%v]", rdeProjector.Spec.RDEId, obj)
+		}
+
+		// Create/Update/Patch the object with the ServerSideApply; types.ApplyPatchType indicates SSA;
+		// fieldManager specifies the owner of the fields getting patched.
+		forceApply := true
+		_, err = dr.Patch(ctx, obj.GetName(), types.ApplyPatchType, data, metav1.PatchOptions{
+			FieldManager: FieldManager,
+			Force:        &forceApply,
+		})
+		if err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "Error patching capi object [%v] via SSA; rdeId: [%s]", rdeProjector.Spec.RDEId, obj)
+		}
```

Comment:
Could we log that we applied the patch of abc object. It is noisy but it can help to know that something is running...maybe at a lower log level.